indico/htdocs/js/indico/jquery/principalfield.js

```diff
@@ -50,15 +50,19 @@
         _add: function _add(people) {
             var self = this;
 
+            function comparePersons(first, second) {
+                /* Compare using EventPerson.id or the e-mail if an id is not available. */
+                if (first.id !== undefined && second.id !== undefined) {
+                    return first.id == second.id;
+                }
```

Comment:
Not really, but since both returns are semantically equally important I think it makes sense this way.